### PR TITLE
fix: look up quorum by actual `quorum_index` not by array position

### PR DESCRIPTION
**Pull request by xdustinface**
We currently look up the quorums by their position in the vector which leads to invalid quroums being selected for IS verification. This PR changes it to look up by actual quorum index. 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved quorum lookup reliability by replacing crash-prone index retrieval with field-based lookup, now returning informative errors when quorums are unavailable instead of panicking.

* **Tests**
  * Added comprehensive test coverage for quorum retrieval, including edge cases and error scenarios to ensure stable behavior.

### CODE REVIEW

File: dash/src/sml/masternode_list_engine/message_request_verification.rs [hunk 1]
```diff
@@ -466,4 +474,86 @@ mod tests {
...
+            include_str!("../../../tests/data/test_DML_diffs/masternode_list_engine.hex");
+        let data = hex::decode(block_hex).expect("decode hex");
+        let mut mn_list_engine: MasternodeListEngine =
+            bincode::decode_from_slice(&data, bincode::config::standard())
+                .expect("expected to decode")
+                .0;
+
+        let lock_data = hex::decode("01018d53e7997ead57409750942af0d5e0aafc06f852a9a52308f4781b6a8220298f00000000c6f9d8c63dd15937ea70aaddb7890daad42c91bf6818e2bf76d183d6f2d9215b4b5f84978fad9dde7ab52bdcc0674be891e9029cc1ef0cb01200000000000000a27c98836c4c04653ab81eb4e07ddfc2c8c2c1036b75247969c05a4f25451cd78913a971f1899d9f2bddec9cf8e0104004f72f20c2856453e5aa3bcd2a8200670ec28feda38f67cc400fc72ef1966956656ec0765478c9d16e9a9e470c07f9ed").expect("expected valid hex");
+        let lock: InstantLock = deserialize(lock_data.as_slice()).expect("expected to deserialize");
+
```
Comment (ZocoLini) [inline dash/src/sml/masternode_list_engine/message_request_verification.rs:491]: extract both strings as constants in the tests module (if decode is defined as constant, extract the whole call but i dont expect it to be const tbh)

File: dash/src/sml/masternode_list_engine/message_request_verification.rs [hunk 2]
```diff
@@ -466,4 +474,86 @@ mod tests {
 
         mn_list_engine.verify_chain_lock(&chain_lock).expect("expected to verify chain lock");
     }
+
+    /// Test that quorums are looked up by their quorum_index field, not by array position.
+    #[test]
+    pub fn is_lock_quorum_lookup_by_quorum_index_not_array_position() {
+        let block_hex =
+            include_str!("../../../tests/data/test_DML_diffs/masternode_list_engine.hex");
+        let data = hex::decode(block_hex).expect("decode hex");
```
Comment (ZocoLini) [inline dash/src/sml/masternode_list_engine/message_request_verification.rs:483]: you can also extract this string though

**Review by ZocoLini — CHANGES_REQUESTED** [review]

File: dash/src/sml/masternode_list_engine/message_request_verification.rs (hunk 1 above)
Comment (xdustinface) [reply dash/src/sml/masternode_list_engine/message_request_verification.rs:491]: This is a fix PR following the existing pattern. Should be cleaned up separately, will just add noise.

Comment (ZocoLini) [reply dash/src/sml/masternode_list_engine/message_request_verification.rs:491]: idk what pattern you are talking about tbh, what ever, not a big deal

**Review by ZocoLini — APPROVED** [review]

File: dash/src/sml/masternode_list_engine/message_request_verification.rs (hunk 1 above)
Comment (xdustinface) [reply dash/src/sml/masternode_list_engine/message_request_verification.rs:491]: Ah haha i though you want to consolidate all other places where those two are used in tests. If thats not it im not sure why you wanted me to move them.
